instrumentation/opentelemetry-instrumentation-fastapi/src/opentelemetry/instrumentation/fastapi/__init__.py

```diff
@@ -246,6 +250,7 @@ def __init__(self, *args, **kwargs):
             client_request_hook=_InstrumentedFastAPI._client_request_hook,
             client_response_hook=_InstrumentedFastAPI._client_response_hook,
             tracer_provider=_InstrumentedFastAPI._tracer_provider,
+            meter_provider=_InstrumentedFastAPI._meter_provider,
```

Comment:
This shouldn't pass the `meter_provider` because the instrumentation scope be will the asgi's `__name__`, and `__version__` which is incorrect. Create a meter here and pass it.

Reply:
How will we pass meter as asgi middleware accepts meter_provider. This change will require changes in Asgi middleware too. Also this same thing is happening in case of tracer_provoder too. Should we change that too?

Reply:
>Also this same thing is happening in case of tracer_provoder too. Should we change that too?

It's a bug with trace provider too but that can be addressed in separate issue. 

>How will we pass meter as asgi middleware accepts meter_provider. This change will require changes in Asgi middleware too.

Yes, it should be changed.

Reply:
added changes as suggested. Please review.

Reply:
This is still not fixed fully.